pkgs/build-support/dotnet/build-dotnet-module/default.nix

```diff
@@ -163,11 +158,10 @@ stdenvNoCC.mkDerivation (args // {
         # That causes conflicts. To circumvent it we remove all occurances of the flag.
         flags =
           let
-            hasRid = flag: lib.any (v: v) (map (rid: lib.hasInfix rid flag) (lib.attrValues dotnet-sdk.runtimeIdentifierMap));
+            isRuntime = flag: lib.hasPrefix "--runtime" flag;
```

Comment:
this seems to be equivalent given what's currently in nixpkgs, and avoids a dependency on dotnetCorePackages, or exposing the list of known rids.

Reply:
I used the solution of relying on `runtimeIdentifiersMap` because there are many ways to pass them to dotnet, I figured this would be a bit more reliable. You could also pass one using an msbuild style flag, or the `-r` prefix, etc.

In my opinion relying on `dotnetCorePackages` is not a big deal since we don't propagate any of its outputs.

Reply:
Without thinking about it, I wrote a derivation that used `[ "--runtime" "linux-x64" ]` (which IMO is more natural than a single string), and it worked fine except for fetch-deps.

There must be a less fragile way of doing this. :thinking:

Reply:
@IvarWithoutBones this is now updated to move all the --runtime flags into the hooks, and remove the special logic here.